mifosng-android/src/main/java/com/mifos/mifosxdroid/online/savingaccountsummary/SavingsAccountSummaryFragment.java

```diff
@@ -456,4 +484,107 @@ public void onDestroyView() {
+            PdfDocument pdfDocument = null;
+            pdfDocument = new PdfDocument();
+            PdfDocument.PageInfo pageInfo = new PdfDocument.PageInfo.Builder(
+                    HeadBitmap.getWidth() + 20, HeadBitmap.getHeight() +
+                    ListBitmap.getHeight() + 20, 1).create();
+            PdfDocument.Page page = pdfDocument.startPage(pageInfo);
+            page.getCanvas().drawBitmap(HeadBitmap, 0, 0, null);
+            page.getCanvas().drawBitmap(ListBitmap, 0, HeadBitmap.getHeight(), null);
+            ListBitmap.recycle();
+            pdfDocument.finishPage(page);
+            String filePath = directory + File.separator + tv_clientName.getText() + ".pdf";
+            File myFile = new File(filePath);
+            try {
+                pdfDocument.writeTo(new FileOutputStream(myFile));
+            } catch (Exception e) {
+                Toast.makeText(getContext(), "Error in saving PDF", Toast.LENGTH_SHORT).show();
+            }
+            pdfDocument.close();
+            Toast.makeText(getContext(), "Pdf saved to " + filePath,
+                    Toast.LENGTH_SHORT).show();
+        } else {
+            Toast.makeText(getContext(),
+                    "This feature is not supported below Android Kitkat(4.4)",
+                    Toast.LENGTH_SHORT).show();
+        }
+    }
+
+    public Bitmap getWholeListViewItemsToBitmap() {
+        ListView listview = lv_Transactions;
+        ListAdapter adapter = listview.getAdapter();
+        int itemscount = adapter.getCount();
+        int allitemsheight = 0;
+        List<Bitmap> bmps = new ArrayList<Bitmap>();
+        for (int i = 0; i < itemscount; i++) {
+            View childView = adapter.getView(i, null, listview);
+            childView.measure(View.MeasureSpec.makeMeasureSpec(listview.getWidth(),
+                    View.MeasureSpec.EXACTLY), View.MeasureSpec.makeMeasureSpec(0,
+                    View.MeasureSpec.UNSPECIFIED));
+            childView.layout(0, 0, childView.getMeasuredWidth(),
+                    childView.getMeasuredHeight());
+            childView.setDrawingCacheEnabled(true);
+            childView.buildDrawingCache();
+            bmps.add(childView.getDrawingCache());
+            allitemsheight += childView.getMeasuredHeight();
+        }
+        Bitmap bigbitmap = Bitmap.createBitmap(listview.getMeasuredWidth(), allitemsheight,
+                Bitmap.Config.ARGB_8888);
+        Canvas bigcanvas = new Canvas(bigbitmap);
+        Paint paint = new Paint();
+        int iHeight = 0;
+        for (int i = 0; i < bmps.size(); i++) {
+            Bitmap bmp = bmps.get(i);
+            bigcanvas.drawBitmap(bmp, 0, iHeight, paint);
+            iHeight += bmp.getHeight();
+            bmp.recycle();
+            bmp = null;
+        }
+        return bigbitmap;
+    }
+
+    public void requestPermission() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
```

Comment:
Avoid using sdk specific permissions.

Reply:
Also practically there are rare chances below kitkat phones, so less chances of occurence and even if this is possible app won't crash. It would inform user.

Reply:
If suppose you have a feature `A` which works on mine device but on yours then would that be a great thing considering user experience. So the point is not about app crashing. 
Let's see what others say about this.

Reply:
I agree with your point danish jamal. I would have agreed about that feature not working for some even if minimum api was Lolipop, if we think practically below kitkat phones rarely used to be 512mb ram ones with low processors so how would they run MifosXDroid app. And even though they run app % given by google using below kitkat is below 1%. On jellybean its not possible to implement this, but we are not thinking for rest 99% devices that can access this feature. Approve this feature if you guys find this appropriate . Thank you : )